src/content/guides/url-parameters/document.ja.mdx

```diff
@@ -0,0 +1,171 @@
+}
+
+export function QuoteView({ id, quote }) {
+  return (
+    <article>
+      <h1>Quote #{id}</h1>
+      <blockquote>
+        {quote}
+      </blockquote>
+    </article>
+  )
+}
+//--- index.js
+import React, { Suspense } from 'react'
+import ReactDOM from 'react-dom'
+import { NotFoundBoundary, Router, View } from 'react-navi'
+import routes from './routes'
+
+ReactDOM.render(
+  <Router routes={routes}>
+    <Suspense fallback={null}>
+      <NotFoundBoundary render={() => <h1>Quote Not Found</h1>}>
+        <View />
+      </NotFoundBoundary>
+    </Suspense>
+  </Router>,
+  document.getElementById('root')
+)
+```
+
+`req.params` には URL parameter だけではなく、URL query string から抽出した情報全てが含まれいます。例えば以下のデモでは `id` と `referrer` パラメーターの両方を `/quotes/3?welcome=true` から取り出す方法を示しています。
+
+```js
+//---
+rightPanel: console
+//--- routes.js
+import React from 'react'
+import { mount, redirect, route } from 'navi'
+import { QuoteView, fetchQuote } from './helpers'
+
+const routes = mount({
+  '/': redirect('/quotes/1?welcome=true'),
+
+  // The ':id' segment will match any value, and its value is made
+  // available within the route at `req.params.id`.
+  '/quotes/:id': route(async req => {
+    console.log('params', req.params)
+
+    let { id, welcome } = req.params
+    let quote = await fetchQuote(id)
+
+    return {
+      view: <QuoteView id={id} quote={quote} welcome={welcome} />,
+    }
+  })
+})
+
+export default routes
+
+//--- helpers.js
+import React from 'react'
+import { NotFoundError } from 'navi'
+
+const quotes = {
+  1: "It is a mistake to think you can solve any major problems just with potatoes.",
+  2: "The ships hung in the sky in much the same way that bricks don’t.",
+  3: "Don't Panic.",
+  42: "“Forty-two,” said Deep Thought, with infinite majesty and calm.",
+}
+
+export async function fetchQuote(id) {
+  let quote = quotes[id]
+
+  return quote ? Promise.resolve(quote) : Promise.reject(new NotFoundError)
+}
+
+export function QuoteView({ id, quote, welcome }) {
+  return (
+    <article>
+      {welcome && <h1>Welcome to QDB</h1>}
+      <h2>Quote #{id}</h2>
+      <blockquote>
+        {quote}
+      </blockquote>
+    </article>
+  )
+}
+//--- index.js
+import React, { Suspense } from 'react'
+import ReactDOM from 'react-dom'
+import { NotFoundBoundary, Router, View } from 'react-navi'
+import routes from './routes'
+
+ReactDOM.render(
+  <Router routes={routes}>
+    <Suspense fallback={null}>
+      <NotFoundBoundary render={() => <h1>Quote Not Found</h1>}>
+        <View />
+      </NotFoundBoundary>
+    </Suspense>
+  </Router>,
+  document.getElementById('root')
+)
+```
+
+URL parameters は JavaScript ルーティングライブラリにおいて非常に標準的な機能ではありますが、URL parameters と asynchronous content を組み合わせることで、莫大な宣言的な力を手に入れることができます。MVC フレームワークにおけるコントローラーのように Navi の route 機能は、ビジネスロジックの実装し、エラーを投げ、View を選択することができます。-- all based on the request. (nakanishi-question: what this request mean? HTTP request?)
```

Comment:
In this case, "Based on the request" means based on the `req` object (which is short for `request`). Perhaps "based on the request object passed to the route function" would be clearer?

Reply:
Thanks! I understand you!